// @bit/danipani.projects.global.social-button/index.js
define(["exports","../../@polymer/lit-element/lit-element.js","../../litelement-fontawesome/index.js","../../@fortawesome/free-brands-svg-icons/index.es.js"],function(_exports,_litElement,_index,_indexEs){"use strict";Object.defineProperty(_exports,"__esModule",{value:!0});_exports.default=void 0;class _default extends _litElement.LitElement{render(){return _litElement.html`
        <style>
          :host {
            --instagram-color: #e4405f;
            --github-color: #333;
            --stackoverflow-color: #f48024;
            --bitbucket-color: #205081;
          }

          div {
            text-align: center;
            font-size: 1.9em;
            --icon-background-color: transparent
          }

          a,
          a:before,
          fontawesome-icon {
            transition: all 0.35s cubic-bezier(0.31, -0.105, 0.43, 1.59);
          }

          a {
            background-color: white;
            display: inline-block;
            width: 3em;
            height: 3em;
            line-height: 3em;
            margin: 0.66em 0.33em;
            position: relative;
            overflow: hidden;
            border-radius: 28%;
            box-shadow: 0 0.16em 0.5em -0.16em rgba(0,0,0,0.1);
            opacity: 0.99;
          }

          a:before {
            top: 90%;
            left: -110%;
            content: '';
            width: 120%;
            height: 120%;
            position: absolute;
            -webkit-transform: rotate(45deg);
                    transform: rotate(45deg);
          }

          fontawesome-icon {
            -webkit-transform: scale(0.8);
            transform: scale(0.8);
          }

          a:hover fontawesome-icon {
            --icon-color: #fff;
            -webkit-transform: scale(1);
                    transform: scale(1);
          }

          a:hover:before {
            top: -10%;
            left: -10%;
          }

          .instagram:before {
            background-color: var(--instagram-color);
          }

          .instagram {
            --icon-color: var(--instagram-color);
          }

          .github:before {
            background-color: var(--github-color);
          }

          .github {
            --icon-color: var(--github-color);
          }

          .stack-overflow:before {
            background-color: var(--stackoverflow-color);
          }

          .stack-overflow {
            --icon-color: var(--stackoverflow-color);
          }

          .bitbucket:before {
            background-color: var(--bitbucket-color);
          }

          .bitbucket {
            --icon-color: var(--bitbucket-color);
          }


        </style>
        <div>
            <a class="instagram"target="_blank" rel="noopener" href="https://www.instagram.com/_danipani_/" aria-label="instagram"><fontawesome-icon  .icon="${_indexEs.faInstagram}"></fontawesome-icon></a>
            <a class="github" target="_blank" rel="noopener" href="https://github.com/DaniPani" aria-label="github"><fontawesome-icon  .icon="${_indexEs.faGithub}"></fontawesome-icon></a>
            <a class="stack-overflow" target="_blank" rel="noopener"  href="https://stackoverflow.com/users/9405824/dani-pani-danipani" aria-label="stack-overflow"><fontawesome-icon .icon="${_indexEs.faStackOverflow}"></fontawesome-icon></a>
            <a class="bitbucket" target="_blank" rel="noopener"  href="https://bitbucket.org/DaniPani/"><fontawesome-icon  .icon="${_indexEs.faBitbucket}" aria-label="bit-bucket"></fontawesome-icon></a>
        </div>
        `}}_exports.default=_default});
