// @bit/danipani.projects.global.scroll-down/index.js
define(["exports","../../@polymer/lit-element/lit-element.js","../../litelement-fontawesome/index.js","../../@fortawesome/free-solid-svg-icons/index.es.js"],function(_exports,_litElement,_index,_indexEs){"use strict";Object.defineProperty(_exports,"__esModule",{value:!0});_exports.default=void 0;class _default extends _litElement.LitElement{render(){return _litElement.html`
        <style>
        :host {
          --icon-color: #fff;
          --icon-background-color: transparent;
        }
        .scroll-down {
          text-align: center;
          cursor: pointer;
          width: 2.875em;
          height: 2.875em;
          line-height: 2.875em;
          border: 0.0625em solid #fff;
          position: relative
        }

        .scroll-down::before {
          content:'';
          position: absolute;
          left: 0;
          width: 2.75em;
          height: 2.75em;
          box-shadow: 0 0 0 0 rgba(255, 255, 255, .1);
          opacity: 0;
          animation: scrollDownAnimation 3s infinite;
        }

        .wrapper {
          width: 10.25em;
          height: 10.25em;
          display: flex;
          align-items: center;
          justify-content: center;
        }

        .scroll-down, .scroll-down::before, .wrapper{
          box-sizing: border-box;
          border-radius: 100%;
        }

        @keyframes scrollDownAnimation {
        0% {
            opacity: 0;
        }
        30% {
            opacity: 1;
        }
        60% {
            box-shadow: 0 0 0 3.75em rgba(255, 255, 255, .1);
            opacity: 0;
        }
        100% {
            opacity: 0;
        }
        }
        </style>
        <div class="wrapper"><div class="scroll-down"><fontawesome-icon .icon="${_indexEs.faAngleDoubleDown}"></fontawesome-icon></div></div>`}}_exports.default=_default});
